src/OrchardCore/OrchardCore.DisplayManagement/Notify/NotifyFilter.cs

```diff
@@ -128,8 +128,7 @@ public async Task OnResultExecutionAsync(ResultExecutingContext filterContext, R
             {
                 foreach (var messageEntry in _existingEntries)
                 {
-                    // Also retrieve the actual zone in case it was only a temporary empty zone created on demand.
-                    zone = await zone.AddAsync(await _shapeFactory.CreateAsync("Message", Arguments.From(messageEntry)));
+                    await zone.AddAsync(await _shapeFactory.CreateAsync("Message", Arguments.From(messageEntry)));
```

Comment:
There's no other similar `AddAsync()` usage?

Reply:
@Piedone  I checked it out, it doesn't seems to be the case, in other places from a given zone only one `AddAsync()` is done. E.g. in the `LayerFilter` when we loop on all widgets, for each widget we always retrieve the related zone from the layout, this before doing `AddAsync()`.

                    var contentZone = layout.Zones[widget.Zone];

                    await contentZone.AddAsync(wrapper, "");

Reply:
Great then!